bundler/spec/bundler/shared_helpers_spec.rb

```diff
@@ -385,6 +385,22 @@
       it_behaves_like "ENV['RUBYOPT'] gets set correctly"
     end
 
+    context "ENV['RUBYOPT'] exists and contains -rbundler/setup with special characters in the path" do
+      source_lib_dir = "/opt/ruby3.3.0-preview2/lib/ruby/3.3.0+0"
```

Comment:
It's a bit non standard to set a variable in this context. Maybe you can group all specs that use the shared specs under a context

```ruby
context "when bundler install path is standard" do
  let(:install_path) { source_lib_dir }

  ### current specs
end

context "when bundler install path contains special characters" do
  let(:install_path) { "/opt/ruby3.3.0-preview2/lib/ruby/3.3.0+0" }

  ### new spec that you added
end
```

And change the shared spec to use `install_path`.

Reply:
Thanks, followed your suggestion.